runtime/jsMain/src/kotlinx/serialization/DynamicObjectSerializer.kt

```diff
@@ -0,0 +1,182 @@
+package kotlinx.serialization
+
+import kotlinx.serialization.internal.StringDescriptor
+import kotlinx.serialization.internal.nullable
+import kotlinx.serialization.json.internal.BEGIN_LIST
+import kotlinx.serialization.json.internal.BEGIN_OBJ
+import kotlinx.serialization.json.internal.END_LIST
+import kotlinx.serialization.json.internal.END_OBJ
+import kotlinx.serialization.modules.EmptyModule
+import kotlinx.serialization.modules.SerialModule
+import kotlinx.serialization.modules.getContextualOrDefault
+
+
+/**
+ * Converts Kotlin data structures to plain Javascript objects
+ *
+ *
+ * Limitations:
+ * * Map keys must be of primitive or enum type
+ * * Custom serializers are not supported
+ * * Enums are serialized as the value of `@SerialName` if present or their name, in that order. Ordinals are not supported.
+ * * There is no analogue to DynamicObjectParsers' useArrayPolymorphism flag
+ *
+ * @param encodeDefaults if true any property which equals exactly its default value will be omitted
```

Comment:
if `encodeDefaults` is true, the default values should be encoded and not omitted.
```suggestion
 * @param encodeDefaults if false any property which equals exactly its default value will be omitted
```